inc/class-module.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * The Module object provides a common interface for registering
+ * modules and carrying out module related tasks.
+ *
+ * @package hm-platform
+ */
+
+namespace HM\Platform;
+
+class Module {
+	/**
+	 * Store of all registered modules.
+	 *
+	 * @var array
+	 */
+	protected static $modules = [];
+
+	/**
+	 * The slug for the module, used to reference individual modules.
+	 *
+	 * @var string
+	 */
+	protected $slug;
+
+	/**
+	 * The directory where the module lives.
+	 *
+	 * @var string
+	 */
+	protected $directory;
+
+	/**
+	 * The human readable title of the module.
+	 *
+	 * @var string
+	 */
+	protected $title;
+
+	/**
+	 * Settings array.
+	 *
+	 * @var array
+	 */
+	protected $settings;
+
+	/**
+	 * Module loader callback function.
+	 *
+	 * @var callable
+	 */
+	protected $loader;
+
+	protected function __construct( string $slug, string $directory, string $title, array $settings, callable $loader ) {
+		$this->slug = $slug;
+		$this->directory = $directory;
+		$this->title = $title;
+		$this->settings = $settings;
+		$this->loader = $loader;
+	}
+
+	/**
+	 * Registers a module with the store.
+	 *
+	 * @param string $slug
+	 * @param string $directory
+	 * @param string $title
+	 * @param array $settings
+	 * @param callable $loader
+	 * @return Module
+	 */
+	public static function register( string $slug, string $directory, string $title, array $settings, callable $loader ) : Module {
```

Comment:
Is `settings` everywhere in this actually "defualt_settings"? Wondering at what stage the config settings get merged in, or if they do at all.

If I want to get a setting for a module, do I go via the `Module` object, or via the `Config` object? Assuming the latter, in which case maybe it makes more sense to call it `default_settings`?

_Or_, is `$settings` meant to be more like the schema for the possible settings for that module?

Reply:
Yeah I think it'll be clearer to rename this param `$default_settings` 👍 

The final settings will indeed come from the config array, something I was doing previously but not implemented here as it should be. Will update to fix that.